### ISSUE TITLE
Craft 2 => 3 Migration : Category assignments duplicated multiple times after migration

### ISSUE THREAD

**Issue by matthisamoto** [craftcms/cms#6895]
### Description
After migrating a site from v. 2.8.0.2 to the current version, entries with multiple nested categories assigned to them get a huge number of duplicates of the same categories assigned to them. I can provide screenshots and other logs/db independently.

The entry we noticed this on has 4 levels deep of categories assigned to it. The top level category is listed 49 times. One of the second level categories is listed 4 time, the other is listed only once. The third and fourth level categories are listed once each. Originally there was only the one top level category with the sub-categories within.

I realize it would be easier to see but I cannot post screenshots of the clients content here.

### Steps to reproduce

1. Have site with entries that have multiple branches of categories assigned
2. Update from Craft 2 => 3
3. Check assigned categories

### Additional info

- Craft version: Original: 2.8.0.2, Updated to: 3.5.11.1
- PHP version: 7.2.32
- Database driver & version: MySQL 5.6.10
- Plugins & versions:
Agnostic Fetch	2.0.4
Amazon S3	1.2.11
Asset Rev	6.0.2
Entries Subset	1.2.2
Entry Count	2.0.3
Inlin	2.1.1
LJ Dynamic Fields	3.0.11
Redactor	2.7.5
Redactor Custom Styles	3.0.4
Reroute	2.0.3
Spoon	3.5.3
Super Table	2.6.2
Table Maker	2.0.1
Typogrify	1.1.18
Video Embed	1.2.2
VZ Address	2.0.2

**Comment by brandonkelly** [comment]
Do the “duplicates” have `draftId` or `revisionId` values in the `elements` table? If so those are the drafts/revisions of the entries, which are now stored as actual elements instead of JSON data within the `entrydrafts` / `entryversions` tables.

**Comment by matthisamoto** [comment]
@brandonkelly All of the duplicated categories on the entry edit screen have the same element ID, which does not have `draftId` or `revisionId`, it is the ID of the Category itself, just over and over.

**Comment by brandonkelly** [comment]
Sorry, I misunderstood. Any chance you can send your Craft 2 and Craft 3 database backups, as well as Composer files, over to support@craftcms.com? We can take a look at it from there.

**Comment by matthisamoto** [comment]
I've sent over these files, thanks!

**Comment by brandonkelly** [comment]
This ended up being due to duplicate data in the `relations` table, even before updating to Craft 3. We’re not sure how that happened in the first place, but the duplication issue goes away when the duplicate data is removed.

**Comment by matthisamoto** [comment]
Appreciate you all taking a look!
